Completed the handle method for incoming post requests

Jonagoo opened the pull request:
Added support for handling incoming HTTP POST requests from the github webhook as well as rudimentary code preparing for future handling of GET requests. Also added some tests for a valid as well as invalid forged request.

This should resolve #5

Jonagoo committed:
Started work on handle method

Certain parts not functional as requestParser class not yet finished.
This is part of solution for #5.

Jonagoo committed:
Fixed issues in executing CI tasks

Changed the code for creating logfile as well as how the execute method is called.
This is part of #5

Jonagoo committed:
Merge branch 'master' into issue5

Get new addition of request parser for work on #5

Jonagoo committed:
Added parsing of request payload to handle method

Handle method now calls relevant methods from requestParser class to get important fields of json payload.
This is part of issue #5 solution

Jonagoo committed:
Fixed syntax issues in handle

Fixed some missing semicolons and parentheses for method calls. Relates to #5

Jonagoo committed:
Fix issue with setting git status

The result of the CI tests was never changed from pending. This affects #5

Jonagoo committed:
Cleaned up parsing of test results

When parsing the test results the program went through an unnecessary loop, now it directly checks for failures from return codes.
Work on #5

Jonagoo committed:
Added support for distinguishing between POST and GET

The handle method now recognizes whether it is handling a POST or GET request, to enable future access to logfiles.
This is part of #5

Jonagoo committed:
Separated handling of post requests into own function

Made handling of post requests its own function to make testing easier.
Part of #5

Jonagoo committed:
Added tests for handling of post requests

Added two tests, one with correct data and one with invalid data.
This should resolve #5

Jonagoo committed:
Added jetty package to running tests

This was required for the CIServerTest to work, as ant couldn't find the abstract class otherwise.
This affects #5

Jonagoo committed:
Removed emailing from handle method

The emailing was causing github to time out as it took too long. As this was an optional requirement, it is removed for the time being.
This should resolve #5

apeinot reviewed (approved):
Your code is fine. Even without the email stuff, we still have the github time out thing. However, I think it's not a big issue as pipeline end and the status is set correctly. We could add a console print when the job is done to see it on the server side.

apeinot69 committed:
Add thread to fix github timeout

Execute pipeline and status in a thread to avoid github timeout
Reenable mail notification
This resolve #5

apeinot commented:
I have resolve the timeout problem by adding thread. Email are on again

Commit:
Merge branch 'master' into issue5

Commit:
Added support for emailing pusher instead of mailbot

Changed the handle of handlePost to enable changing recipient of email containing results of tests.
This should resolve #5

Commit:
Added documentation for changes to handlePost

The email parameter wasn't included in the documentation.
This should resolve #5

Commit:
Changed email recipient back to mailbot

When the pushers github email was set to private the email field in the json wasa noreply email, so it was decided to send it to mailbot instead.
This should resolve #5
